mllib/src/main/scala/org/apache/spark/ml/param/shared/SharedParamsCodeGen.scala

```diff
@@ -78,7 +78,24 @@ private[shared] object SharedParamsCodeGen {
       ParamDesc[String]("solver", "the solver algorithm for optimization. If this is not set or " +
         "empty, default value is 'auto'.", Some("\"auto\"")))
 
-    val code = genSharedParams(params)
+    // scalastyle:off
+    val extras: Seq[String] = Seq(
+      """
+        |private[ml] trait HasInitialModel[T <: Model[T]] extends Params {
+        |
+        |  /**
+        |   * Param for initial model of warm start.
```

Comment:
I prefer to put them in a separate place for this PR because I don't want to get involved in too many features in a single PR. Let's add follow-up JIRAs for other features.